replicator/src/test/scala/com/evolutiongaming/kafka/journal/replicator/BatchSpec.scala

```diff
@@ -41,164 +41,170 @@ class BatchSpec extends AnyFunSuite with Matchers {
+          delete(offset = 4, to    = 2, origin = "origin2"),
         ),
-        List(appends(3, append(offset = 3, seqNr = 2)), deletes(offset = 4, seqNr = 2, origin = "origin1")),
+        List(appends(3, append(offset = 3, seqNr = 2)), deletes(offset = 4, to = 2, origin = "origin2")),
       ),
       (
         Nel.of(
           append(offset = 1, seqNr = 1),
-          delete(offset = 2, seqNr = 1, origin = "origin"),
+          delete(offset = 2, to    = 1, origin = "origin"),
           append(offset = 3, seqNr = 2),
-          delete(offset = 4, seqNr = 2),
+          delete(offset = 4, to    = 2),
         ),
-        List(appends(3, append(offset = 3, seqNr = 2)), deletes(offset = 4, seqNr = 2, origin = "origin")),
+        List(appends(3, append(offset = 3, seqNr = 2)), deletes(offset = 4, to = 2)),
       ),
       (
         Nel.of(
           append(offset = 1, seqNr = 1),
           append(offset = 2, seqNr = 2),
-          delete(offset = 3, seqNr = 1, origin = "origin1"),
-          delete(offset = 4, seqNr = 2, origin = "origin2"),
+          delete(offset = 3, to    = 1, origin = "origin1"),
+          delete(offset = 4, to    = 2, origin = "origin2"),
         ),
-        List(appends(2, append(offset = 2, seqNr = 2)), deletes(offset = 4, seqNr = 2, origin = "origin1")),
+        List(appends(2, append(offset = 2, seqNr = 2)), deletes(offset = 4, to = 2, origin = "origin2")),
       ),
       (
         Nel.of(
           append(offset = 1, seqNr = 1),
           append(offset = 2, seqNr = 2),
-          delete(offset = 3, seqNr = 1),
-          delete(offset = 4, seqNr = 2, origin = "origin"),
+          delete(offset = 3, to    = 1),
+          delete(offset = 4, to    = 2, origin = "origin"),
         ),
         List(
           appends(offset = 2, append(offset = 2, seqNr = 2)),
-          deletes(offset = 4, seqNr = 2, origin = "origin"),
+          deletes(offset = 4, to = 2, origin = "origin"),
         ),
       ),
-      (Nel.of(delete(offset = 2, seqNr = 1), delete(offset = 3, seqNr = 2)), List(deletes(offset = 3, seqNr = 2))),
+      (Nel.of(delete(offset = 2, to = 1), delete(offset = 3, to = 2)), List(deletes(offset = 3, to = 2))),
       (
-        Nel.of(delete(offset = 2, seqNr = 2, origin = "origin"), delete(offset = 3, seqNr = 1)),
-        List(deletes(offset = 3, seqNr = 2, origin = "origin")),
+        Nel.of(delete(offset = 2, to = 2, origin = "origin"), delete(offset = 3, to = 1)),
+        List(deletes(offset = 2, to = 2, origin = "origin")),
       ),
       (
         Nel.of(
           mark(offset   = 2),
-          delete(offset = 3, seqNr = 1, origin = "origin"),
+          delete(offset = 3, to = 1, origin = "origin"),
           mark(offset   = 4),
-          delete(offset = 5, seqNr = 2),
+          delete(offset = 5, to = 2),
           mark(offset   = 6),
         ),
-        List(deletes(offset = 6, seqNr = 2, origin = "origin")),
+        List(deletes(offset = 5, to = 2)),
```

Comment:
here we lose the `origin`

Reply:
for all above case with "here we lose the `origin`", very possibly we lose `version` too.

basically it means that `origin` and `version`, probably, will get set by one of following `append` actions, instead of [possibly] first-journal-creating `delete` action